pymetabiosis/numpy_convert.py

```diff
@@ -24,3 +26,55 @@ def call_direct(f):
 
     if hasattr(numpy, "float128"):
         cpy_to_pypy_converters.update({numpy.float128._cpyobj: call_direct(builtin.float)})
+
+
+def register_pypy_cpy_ndarray_converters():
+    ''' Convert numpy types from PyPy numpy to CPython numpy, and back.
+    '''
+    import numpy
+    cpython_numpy = import_module("numpy")
+    pypy_to_cpy_converters[numpy.ndarray] = convert_ndarray
+    cpy_to_pypy_converters[cpython_numpy.ndarray._cpyobj] = \
+        convert_from_ndarray
+
+_convert_ndarray = applevel('''
+import numpy
+import ctypes
+def _convert_ndarray(shape, dtype, size, addr):
+    return numpy.ndarray(
+        shape=shape,
+        dtype=dtype,
+        buffer=(ctypes.c_char*size).from_address(addr))
+return _convert_ndarray
+''', noconvert=True)
+
+def convert_ndarray(obj):
+    w = _convert_ndarray(
+        obj.shape,
+        obj.dtype.name,
+        len(obj.data),
+        obj.__array_interface__["data"][0])
+    return ffi.gc(w._cpyobj, lib.Py_DECREF)
```

Comment:
Reference count of w._cpyobj just before the return is 2, (as expected for a freshly created object I think?), and I was unable to make anything crash. But overall I am not sure, maybe we need a robust way to check that we do not leak memory and keep proper reference count.